xarray/tests/test_missing.py

```diff
@@ -0,0 +1,441 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import numpy as np
+import pandas as pd
+import pytest
+import itertools
+
+import xarray as xr
+
+from xarray.core.missing import (NumpyInterpolator, ScipyInterpolator,
+                                 SplineInterpolator)
+from xarray.core.pycompat import dask_array_type
+
+from xarray.tests import (assert_equal, assert_array_equal, raises_regex,
+                          requires_scipy, requires_bottleneck, requires_dask,
+                          requires_np112)
+
+
+@pytest.fixture
+def da():
+    return xr.DataArray([0, np.nan, 1, 2, np.nan, 3, 4, 5, np.nan, 6, 7],
+                        dims='time')
+
+
+@pytest.fixture
+def ds():
+    ds = xr.Dataset()
+    ds['var1'] = xr.DataArray([0, np.nan, 1, 2, np.nan, 3, 4, 5, np.nan, 6, 7],
+                              dims='time')
+    ds['var2'] = xr.DataArray([10, np.nan, 11, 12, np.nan, 13, 14, 15, np.nan,
+                               16, 17], dims='x')
+    return ds
+
+
+def make_interpolate_example_data(shape, frac_nan, seed=12345,
+                                  non_uniform=False):
+    rs = np.random.RandomState(seed)
+    vals = rs.normal(size=shape)
+    if frac_nan == 1:
+        vals[:] = np.nan
+    elif frac_nan == 0:
+        pass
+    else:
+        n_missing = int(vals.size * frac_nan)
+
+        ys = np.arange(shape[0])
+        xs = np.arange(shape[1])
+        if n_missing:
+            np.random.shuffle(ys)
+            ys = ys[:n_missing]
+
+            np.random.shuffle(xs)
+            xs = xs[:n_missing]
+
+            vals[ys, xs] = np.nan
+
+    if non_uniform:
+        # construct a datetime index that has irregular spacing
+        deltas = pd.TimedeltaIndex(unit='d', data=rs.normal(size=shape[0],
+                                                            scale=10))
+        coords = {'time': (pd.Timestamp('2000-01-01') + deltas).sort_values()}
+    else:
+        coords = {'time': pd.date_range('2000-01-01', freq='D',
+                  periods=shape[0])}
+    da = xr.DataArray(vals, dims=('time', 'x'), coords=coords)
+    df = da.to_pandas()
+
+    return da, df
+
+
+@requires_np112
+@requires_scipy
+def test_interpolate_pd_compat():
+    shapes = [(8, 8), (1, 20), (20, 1), (100, 100)]
+    frac_nans = [0, 0.5, 1]
+    methods = ['linear',  'nearest', 'zero', 'slinear', 'quadratic', 'cubic']
+
+    for (shape, frac_nan, method) in itertools.product(shapes, frac_nans,
+                                                       methods):
+
+        da, df = make_interpolate_example_data(shape, frac_nan)
+
+        for dim in ['time', 'x']:
+            actual = da.interpolate_na(method=method, dim=dim)
+            expected = df.interpolate(method=method, axis=da.get_axis_num(dim))
+            np.testing.assert_allclose(actual.values, expected.values)
```

Comment:
Perennial note for future tests: `xarray.tests.assert_close`

Reply:
expected is a Pandas.DataFrame so I don't think this will work with `xarray.tests.assert_close`